lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems      = ExtManagementSystem.find(ems_id)
+all_emss = [ems] + ems.child_managers
+ems_opts = all_emss.map do |e|
+  e.attributes.merge(
+    "endpoints"       => e.endpoints,
+    "authentications" => e.authentications
+  )
+end
+
+provider_opts = {
+  :ems       => ems_opts,
+  :messaging => MiqQueue.messaging_client_options,
+  :settings  => Settings.to_h
```

Comment:
This is fine for now - wonder if we can find a way to only give a subset of this later.  Also, we might still encrypt some settings, such as proxy passwords, so they might need decryption, but we can do that in a followup.

Reply:
Yeah I was considering worker_settings but we actually use a decent number of settings that are outside of that (logging, proxy, ems/*, etc...)
When we got to "reloading settings" and when to restart workers I was considering something like "have the worker class tell us which settings keys they care about".
We could also have a command to load Settings and dump to stdout that can be called which could also be used by the worker in the "reload" case.
I can drop the `Settings.to_h` here for now if we want to tackle that in a better way later, up to you.

Reply:
I'm fine with the hashifying all of the settings for now.  But it should be this instead, because .to_h only hashifies one level deep, whereas .to_hash is the whole thing.

```suggestion
  :settings  => Settings.to_hash
```

Reply:
Done, we are .to_json'ing the whole thing anyway but this works